### PR TITLE
applications: asset_tracker_v2: Update `roam` and `dev` objects

### PR DESCRIPTION
**Pull request by simensrostad**
Update `roam` and `dev` objects:

- Move band and network mode from `dev` to `roam`.

*The reported network mode is now the mode that is actually used by
the modem in the LTE connection, not the network modes that the modem
is configured to use.*

- Add IMEI to `dev` object.

Fixes CIA-463, NCSIDB-641, and CIA-368

### CODE REVIEW

**Comment by simensrostad** [comment]
Had some offline discussions regarding this PR and decided to put it on hold until after the release. The PR will be updated to make the application publish the LTE mode that the modem actually connects with. The parameter will also be moved from (deviceInfo / dev) to (networkInfo / roam).

**Comment by coderbyheart** [comment]
See https://github.com/NordicSemiconductor/asset-tracker-cloud-docs/issues/328
and https://github.com/NordicSemiconductor/asset-tracker-cloud-docs/issues/327

**Comment by jtguggedal** [comment]
> Had some offline discussions regarding this PR and decided to put it on hold until after the release. The PR will be updated to make the application publish the LTE mode that the modem actually connects with. The parameter will also be moved from (deviceInfo / dev) to (networkInfo / roam).

Then we should probably add it as a known issue that the mode that is reported in some cases is wrong

**Comment by simensrostad** [comment]
Blocked by https://github.com/nrfconnect/sdk-nrf/pull/6405

**Review by jtguggedal — APPROVED** [review]

File: applications/asset_tracker_v2/src/modules/Kconfig.app_module [hunk 1]
```diff
@@ -8,7 +8,7 @@ menu "Application control module"
 
 config APP_REQUEST_GNSS_ON_INITIAL_SAMPLING
 	bool "Request GNSS on initial data sampling after connection"
-	default y if GPS_MODULE
+	default y if GNSS_MODULE
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/modules/Kconfig.app_module:11]:
```suggestion
	default y if GPS_MODULE
```

Addressed in #6496, an-it's a non-related change.

**Comment by simensrostad** [comment]
@coderbyheart Addressed your comments, thanks.

**Review by coderbyheart — APPROVED** [review]

**Pull request merged**